capstone/src/main/java/com/google/sps/Objects/response/MemberResponse.java

```diff
@@ -58,6 +58,30 @@ public static MemberResponse fromEntity(Entity entity, boolean includeBadges) {
     return new MemberResponse(profilePic, firstName, lastName, badges, userId);
   }
 
+  /**
+   * Checks if two Member Response Objects are equal
+   *
+   * @param other Member Response object 
+   */
+  @Override
+  public boolean equals(Object other) {
+    if (other == null) return false;
+    if (other == this) return true;
+    if (!(other instanceof MemberResponse)) return false;
+    MemberResponse memResponse = (MemberResponse) other;
+    if (memResponse.badges != null && memResponse.badges.size() > 0) {
+      return profilePic.equals(memResponse.profilePic) &&
+        firstName.equals(memResponse.firstName) &&
+        lastName.equals(memResponse.lastName) &&
+        userId.equals(memResponse.userId) &&
+        badges.containsAll(memResponse.badges);
+    }
+    return profilePic.equals(memResponse.profilePic) &&
+      firstName.equals(memResponse.firstName) &&
+      lastName.equals(memResponse.lastName) &&
+      userId.equals(memResponse.userId);
```

Comment:
If you end up going the JSON route, you may want to create temporary objects where the badges field would ALWAYS be an array (instead of null) then convert them to json, then string compare

Reply:
If I retrieve an entity from the database with an empty list of badges, the property badges of the entity would return null. Added this check in case I want to compare two objects after getting the entity from the database.